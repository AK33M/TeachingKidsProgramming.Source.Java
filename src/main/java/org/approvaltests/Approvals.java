package org.approvaltests;

import java.awt.Component;
import java.awt.Image;
import java.awt.image.BufferedImage;
import java.io.File;
import java.util.Arrays;
import java.util.Map;

import org.approvaltests.ReporterFactory.FileTypes;
import org.approvaltests.approvers.FileApprover;
import org.approvaltests.core.ApprovalFailureOverrider;
import org.approvaltests.core.ApprovalFailureReporter;
import org.approvaltests.core.ApprovalWriter;
import org.approvaltests.namer.ApprovalNamer;
import org.approvaltests.namer.JUnitStackTraceNamer;
import org.approvaltests.writers.ApprovalTextWriter;
import org.approvaltests.writers.ApprovalXmlWriter;
import org.approvaltests.writers.ComponentApprovalWriter;
import org.approvaltests.writers.FileApprovalWriter;
import org.approvaltests.writers.ImageApprovalWriter;
import org.lambda.functions.Function1;

import org.teachingkidsprogramming.util.ArrayUtils;
import org.teachingkidsprogramming.util.ObjectUtils;
import com.spun.util.StringUtils;

public class Approvals {
	public static void verify(String response) throws Exception {
		verify(new ApprovalTextWriter(response, "txt"), FileTypes.Text);
	}

	public static void verify(Object o) throws Exception {
		verify("" + o);
	}

	public static <T> void verifyAll(String label, T[] array) throws Exception {
		verify(new ApprovalTextWriter(StringUtils.toString(label, array), "txt"),
				FileTypes.Text);
	}

	public static <T> void verifyAll(String header, String label, T[] array)
			throws Exception {
		verify(new ApprovalTextWriter(formatHeader(header)
				+ StringUtils.toString(label, array), "txt"), FileTypes.Text);
	}

	public static <T> void verifyAll(T[] values, Function1<T, String> f1) {
		String text = ArrayUtils.toString(values, f1);
		verify(new ApprovalTextWriter(text, "txt"), FileTypes.Text);
	}

	public static <T> void verifyAll(String header, T[] values,
			Function1<T, String> f1) {
		verifyAll(header, Arrays.asList(values), f1);
	}

	public static <T> void verifyAll(String header, Iterable<T> array,
			Function1<T, String> f1) {
		String text = formatHeader(header) + ArrayUtils.toString(array, f1);
		verify(new ApprovalTextWriter(text, "txt"), FileTypes.Text);
	}

	private static String formatHeader(String header) {
		return StringUtils.isEmpty(header) ? "" : header + "\r\n\r\n\r\n";
	}

	public static <T> void verifyAll(String label, Iterable<T> array)
			throws Exception {
		verify(new ApprovalTextWriter(StringUtils.toString(label, array), "txt"),
				FileTypes.Text);
	}

	public static <T> void verifyAll(String header, String label,
			Iterable<T> array) throws Exception {
		verify(new ApprovalTextWriter(formatHeader(header)
				+ StringUtils.toString(label, array), "txt"), FileTypes.Text);
	}

	public static void verify(Component c) throws Exception {
		verify(new ComponentApprovalWriter(c), FileTypes.Image);
	}

	public static void verifyHtml(String response) throws Exception {
		verify(new ApprovalTextWriter(response, "html"), FileTypes.Html);
	}

	public static void verify(File generateFile) {
		verify(new FileApprovalWriter(generateFile), FileTypes.File);
	}

	public static void verify(Image image) {
		// approve(ImageWriter.toBufferedImage(image), createApprovalNamer());
	}

	public static void verify(BufferedImage bufferedImage) {
		verify(new ImageApprovalWriter(bufferedImage), FileTypes.Image);
	}

	public static void verify(ApprovalWriter writter, ApprovalNamer namer,
			ApprovalFailureReporter reporter) {
		verify(new FileApprover(writter, namer), reporter);
	}

	public static void verify(ApprovalWriter writter, String fileType) {
		verify(writter, createApprovalNamer(), ReporterFactory.get(fileType));
	}

	public static void verifyXml(String xml) throws Exception {
		verify(new ApprovalXmlWriter(xml), FileTypes.Text);
	}

	public static void verify(FileApprover approver,
			ApprovalFailureReporter reporter) {
		try {
			if (!approver.approve()) {
				boolean passed = false;
				if (reporter instanceof ApprovalFailureOverrider) {
					passed = approver
							.askToChangeReceivedToApproved((ApprovalFailureOverrider) reporter);
				}
				if (!passed) {
					approver.reportFailure(reporter);
					approver.fail();
				} else {
					approver.cleanUpAfterSuccess(reporter);
				}
			} else {
				approver.cleanUpAfterSuccess(reporter);
			}
		} catch (Exception e) {
			throw ObjectUtils.throwAsError(e);
		}
	}

	public static void verify(Map map) throws Exception {
		verify(new ApprovalTextWriter(StringUtils.toString(map), "txt"),
				FileTypes.Text);
	}

	public static ApprovalNamer createApprovalNamer() {
		return new JUnitStackTraceNamer();
	}

	private static void approve(BufferedImage bufferedImage, ApprovalNamer namer) {
		verify(new ImageApprovalWriter(bufferedImage), FileTypes.Image);
	}

//	private static String getFileNameList(List<File> mismatched) {
//		return Query.select(mismatched,
//				new F1<File, String>(mismatched.get(1)) {
//					{
//						ret(a.getName());
//					}
//				}).toString();
//	}
}