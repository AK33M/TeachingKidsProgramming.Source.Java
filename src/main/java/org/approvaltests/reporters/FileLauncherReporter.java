package org.approvaltests.reporters;


import org.approvaltests.core.ApprovalFailureReporter;

import org.teachingkidsprogramming.util.TestUtils;
import com.spun.util.io.FileUtils;

public class FileLauncherReporter implements ApprovalFailureReporter
{
  //@Override
  public void report(String received, String approved) throws Exception
  {
    if (FileUtils.isNonEmptyFile(approved))
    {
      TestUtils.displayFile(approved);
    }
    TestUtils.displayFile(received);
  }
}
