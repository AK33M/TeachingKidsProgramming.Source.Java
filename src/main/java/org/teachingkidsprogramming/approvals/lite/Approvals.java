package org.teachingkidsprogramming.approvals.lite;

import org.teachingkidsprogramming.approvals.lite.ReporterFactory.FileTypes;
import org.teachingkidsprogramming.approvals.lite.namer.ApprovalNamer;
import org.teachingkidsprogramming.approvals.lite.namer.JUnitStackTraceNamer;
import org.approvaltests.writers.ApprovalTextWriter;
import org.approvaltests.writers.ImageApprovalWriter;
import org.lambda.functions.Function1;
import org.teachingkidsprogramming.util.ArrayUtils;
import org.teachingkidsprogramming.util.ObjectUtils;
import org.teachingkidsprogramming.util.StringUtils;

import java.awt.image.BufferedImage;
import java.util.Arrays;

public class Approvals {
	public static void verify(String response) throws Exception {
		verify(new ApprovalTextWriter(response, "txt"), FileTypes.Text);
	}

    public static <T> void verifyAll(String header, T[] values,
			Function1<T, String> f1) {
		verifyAll(header, Arrays.asList(values), f1);
	}

	public static <T> void verifyAll(String header, Iterable<T> array,
			Function1<T, String> f1) {
		String text = formatHeader(header) + ArrayUtils.toString(array, f1);
		verify(new ApprovalTextWriter(text, "txt"), FileTypes.Text);
	}

	private static String formatHeader(String header) {
		return StringUtils.isEmpty(header) ? "" : header + "\r\n\r\n\r\n";
	}

    public static void verifyHtml(String response) throws Exception {
		verify(new ApprovalTextWriter(response, "html"), FileTypes.Html);
	}

    public static void verify(BufferedImage bufferedImage) {
		verify(new ImageApprovalWriter(bufferedImage), FileTypes.Image);
	}

	public static void verify(ApprovalWriter writter, ApprovalNamer namer,
			ApprovalFailureReporter reporter) {
		verify(new FileApprover(writter, namer), reporter);
	}

	public static void verify(ApprovalWriter writter, String fileType) {
		verify(writter, createApprovalNamer(), ReporterFactory.get(fileType));
	}

    public static void verify(FileApprover approver,
			ApprovalFailureReporter reporter) {
		try {
			if (!approver.approve()) {
				boolean passed = false;
				if (reporter instanceof ApprovalFailureOverrider) {
					passed = approver
							.askToChangeReceivedToApproved((ApprovalFailureOverrider) reporter);
				}
				if (!passed) {
					approver.reportFailure(reporter);
					approver.fail();
				} else {
					approver.cleanUpAfterSuccess(reporter);
				}
			} else {
				approver.cleanUpAfterSuccess(reporter);
			}
		} catch (Exception e) {
			throw ObjectUtils.throwAsError(e);
		}
	}

    public static ApprovalNamer createApprovalNamer() {
		return new JUnitStackTraceNamer();
	}

}