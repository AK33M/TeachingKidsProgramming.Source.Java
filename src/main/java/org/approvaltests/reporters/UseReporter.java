package org.approvaltests.reporters;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import org.teachingkidsprogramming.approvals.lite.ApprovalFailureReporter;

@Retention(RetentionPolicy.RUNTIME)
public @interface UseReporter {
  Class<? extends ApprovalFailureReporter>[] value();
}
