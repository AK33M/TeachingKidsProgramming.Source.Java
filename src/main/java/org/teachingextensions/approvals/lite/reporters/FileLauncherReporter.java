package org.teachingextensions.approvals.lite.reporters;


import org.teachingextensions.approvals.lite.ApprovalFailureReporter;

import org.teachingkidsprogramming.util.TestUtils;
import org.teachingkidsprogramming.util.io.FileUtils;

public class FileLauncherReporter implements ApprovalFailureReporter
{
  //@Override
  public void report(String received, String approved) throws Exception
  {
    if (FileUtils.isNonEmptyFile(approved))
    {
      TestUtils.displayFile(approved);
    }
    TestUtils.displayFile(received);
  }
}
