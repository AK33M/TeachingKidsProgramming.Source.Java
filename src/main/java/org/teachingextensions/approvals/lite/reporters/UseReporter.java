package org.teachingextensions.approvals.lite.reporters;

import org.teachingextensions.approvals.lite.ApprovalFailureReporter;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface UseReporter {
    Class<? extends ApprovalFailureReporter>[] value();
}
