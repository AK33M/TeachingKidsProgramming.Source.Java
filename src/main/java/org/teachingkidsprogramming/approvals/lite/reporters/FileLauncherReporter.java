package org.teachingkidsprogramming.approvals.lite.reporters;


import org.teachingkidsprogramming.approvals.lite.ApprovalFailureReporter;

import org.teachingkidsprogramming.util.TestUtils;
import org.teachingkidsprogramming.util.io.FileUtils;

public class FileLauncherReporter implements ApprovalFailureReporter
{
  //@Override
  public void report(String received, String approved) throws Exception
  {
    if (FileUtils.isNonEmptyFile(approved))
    {
      TestUtils.displayFile(approved);
    }
    TestUtils.displayFile(received);
  }
}
