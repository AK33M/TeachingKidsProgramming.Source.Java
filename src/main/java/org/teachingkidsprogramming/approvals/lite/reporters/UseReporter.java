package org.teachingkidsprogramming.approvals.lite.reporters;

import org.teachingkidsprogramming.approvals.lite.ApprovalFailureReporter;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface UseReporter {
    Class<? extends ApprovalFailureReporter>[] value();
}
