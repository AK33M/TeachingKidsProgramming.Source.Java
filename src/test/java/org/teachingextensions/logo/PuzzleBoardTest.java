package org.teachingextensions.logo;

import org.junit.Test;
import org.teachingextensions.approvals.lite.Approvals;
import org.teachingextensions.approvals.lite.reporters.FileLauncherReporter;
import org.teachingextensions.approvals.lite.reporters.UseReporter;

@UseReporter(FileLauncherReporter.class)
public class PuzzleBoardTest {
  /**
   * A puzzle board shows the initial puzzle.
   */
  @Test
  public void show_initial_puzzle() throws Exception {
    int[] cells = {0, 1, 2, 3, 4, 5, 6, 7, 8};
    PuzzleBoard board = new PuzzleBoard(new Puzzle(cells), null);
    PuzzleWindow window = new PuzzleWindow();
    window.add(board);
    Approvals.verify(window);
  }

  /**
   * A puzzle board shows the provided puzzle
   */
  @Test
  public void show_provided_puzzle() throws Exception {
    int[] cells = {0, 1, 2, 3, 4, 5, 6, 8, 7};
    PuzzleBoard board = new PuzzleBoard(new Puzzle(cells), null);
    PuzzleWindow window = new PuzzleWindow(board);
    Approvals.verify(window);
  }
}
