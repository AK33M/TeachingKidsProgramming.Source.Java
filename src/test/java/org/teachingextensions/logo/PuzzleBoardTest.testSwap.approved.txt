PuzzleBoard
Tile{35, 35}, Tile{162, 35}, Tile{289, 35}
Tile{35, 162}, Tile{162, 162}, Tile{289, 162}
Tile{35, 289}, Tile{289, 289}, null
